src/sentry/tagstore/snuba/backend.py

```diff
@@ -157,21 +162,50 @@ def __get_tag_keys_for_projects(
         limit=1000,
         keys=None,
         include_values_seen=True,
+        use_cache=True,
         **kwargs
     ):
-        filters = {"project_id": projects}
+        filters = {"project_id": sorted(projects)}
         if environments:
-            filters["environment"] = environments
+            filters["environment"] = sorted(environments)
         if group_id is not None:
             filters["group_id"] = [group_id]
         if keys is not None:
-            filters["tags_key"] = keys
+            filters["tags_key"] = sorted(keys)
         aggregations = [["count()", "", "count"]]
 
         if include_values_seen:
             aggregations.append(["uniq", "tags_value", "values_seen"])
         conditions = []
 
+        should_cache = (
+            random.random() <= options.get("snuba.tagstore.cache-tagkeys-rate")
+            and use_cache
+            and group_id is None
+        )
```

Comment:
Our metrics also sample at 10% by default. Keep the double sampling rates in mind when looking at your results.